agent/bootstrap.go

```diff
@@ -116,19 +120,28 @@ func (boot bootstrap) Run() (err error) {
 }
 
 func (boot bootstrap) setUserPasswords(env boshsettings.Env) error {
-	password := env.GetPassword()
+	password := env.GetUserPassword()
 	if password == "" {
 		return nil
 	}
 
-	err := boot.platform.SetUserPassword(boshsettings.RootUsername, password)
+	err := boot.platform.SetUserPassword(boshsettings.VCAPUsername, password)
 	if err != nil {
-		return bosherr.WrapError(err, "Setting root password")
+		return bosherr.WrapError(err, "Setting vcap password")
 	}
 
-	err = boot.platform.SetUserPassword(boshsettings.VCAPUsername, password)
+	return nil
+}
+
+func (boot bootstrap) setRootPasswords(env boshsettings.Env) error {
+	password := env.GetRootPassword()
+	if password == "" {
+		return nil
```

Comment:
i dont think we should check for emptiness. if the goal is to stop resetting root password then we should just add a boolean configuration `keep_root_password` on BoshEnv so that root password is not changed.